internal/compiler-interface/src/main/java/xsbti/AnalysisCallback2.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright Scala Center, Lightbend, and Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package xsbti;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extension to AnalsysisCallback.
+ * This is a compatibility layer created for Scala 3.x compilers.
+ * Even though compiler-interface is NOT intended to be a public API,
+ * Scala 3 compiler publishes their own compiler bridge against Zinc de jour.
+ * By having this interface, they can test if `callback` supports AnalysisCallback2.
```

Comment:
How do you envision this working? If the dotty bridge starts doing `isInstanceOf[AnalysisCallback2]` we'll get a class not found exception if we're running against an older sbt/zinc, so we'd need to catch the exception I guess?

Reply:
Yea, I guess something like that. I figured it's better than just adding a new method.

Reply:
also we have methodhandles

Reply:
I guess we can work with that, but it'd be great if we could coordinate on this sort of things before sbt releases in the future so we can take the time to come up with the best way forward, for example if we had gone with a `problem` that takes an `xsbti.Problem` we could have ensured we don't need another interface like this next time we add a field to `xsbti.Problem`.

Reply:
Yea, I wanted to unblock the development on this, but I guess in the hindsight I should've waited for reviews.